src/main/java/duke/storage/LocalStorage.java

```diff
@@ -0,0 +1,143 @@
+package duke.storage;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.DateTimeException;
+import java.time.LocalDate;
+
+import duke.exception.DukeException;
+import duke.exception.InvalidArgumentException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Local storage to store the data from local file.
+ */
+public class LocalStorage {
+    private File file;
+
+    /**
+     * Constructor to create a local storage using the data from local file.
+     * @param path
+     */
+    public LocalStorage(String path) {
+        File localFile = readFile(path);
+        this.file = localFile;
+    }
+
+    /**
+     * Read from file in local duke.storage.
+     * @param path Path to the file in the local duke.storage.
+     * @return File obtained from local storage
+     */
+    public static File readFile(String path) {
+        File file = new File(path);
+        file.getParentFile().mkdirs();
+
+        if (file.exists()) {
+            return file;
+        }
+
+        try {
+            file.createNewFile();
+        } catch (IOException io_error) {
+            io_error.printStackTrace();
+        }
+
+        return file;
+    }
+
+    /**
+     * Load tasks from file.
+     * @param tasks List to add the tasks read from file.
+     */
+    public void loadTasks(TaskList tasks) {
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(this.file));
+            String line = reader.readLine();
+
+            while (line != null) {
+                try {
+                    String[] args = line.split("\\|");
+                    String taskType = args[0].strip();
+                    String taskStatus = args[1].strip();
+                    String taskDesc = args[2].strip();
+                    switch (taskType) {
+                    case "T":
+                        Todo todo = new Todo(taskDesc);
+                        if (taskStatus.equals("1")) {
+                            todo.markComplete();
+                        }
+                        tasks.add(todo);
+                        break;
+                    case "D":
+                        String dueDate = args[3].trim();
+                        try {
+                            LocalDate formattedDueDate = LocalDate.parse(dueDate);
+                            Deadline deadline = new Deadline(taskDesc, formattedDueDate);
+                            if (taskStatus.equals("1")) {
+                                deadline.markComplete();
+                            }
+                            tasks.add(deadline);
+                            break;
+                        } catch (DateTimeException error) {
+                            throw new InvalidArgumentException("Wrong date format! Please follow the format "
```

Comment:
Similar to above, since this string is used in multiple lines for the same exception, might be good to store it in the exception itself